Mathlib/Data/Set/ToFunction.lean

```diff
@@ -0,0 +1,64 @@
+/-
+Copyright (c) 2024 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak
+-/
+import Mathlib.Data.Set.Defs
+import Mathlib.Logic.ExistsUnique
+
+/-!
+# Set to function
+
+This file provides API for sets that define functions.
+
+## Main definitions
+- `Set.asPartialFun` converts `Set (α × α)` to `α → Option α`.
+- `Set.asFun` converts `Set (α × α)` to `α → α` if possible.
+
+-/
+
+variable {α : Type*}
+
+namespace Set
+
+section set_as_partial_function
+
+/-- Does given set on `α × α` represent a partial function? -/
+def isPartialFun (X : Set (α × α)) : Prop :=
+  ∀ x y z : α, (x, y) ∈ X ∧ (x, z) ∈ X → y = z
```

Comment:
You can write this as
```suggestion
  ∀ x : α, {y : α | (x, y) ∈ X}.Subsingleton
```
I believe you can even get rid of the type annotations if you want.